src/lnd-setup/generate-wallet-unlocker-client.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Wallet Unlocker Client Module
+ * @module src/lnd-setup/generate-wallet-unlocker-client
+ */
+
+const grpc = require('grpc')
+const loadProto = require('../utils/load-proto')
+const fs = require('fs')
+
+/**
+ * Generates a lnrpc.WalletUnlocker client which is only used on initialization of the LND
+ * node.
+ *
+ * @function
+ * @param {LndEngine} - engine
+ * @param {String} engine.host
+ * @param {String} engine.protoPath
+ * @param {String} engine.tlsCertPath
+ * @returns {grpc.Client} lnrpc WalletUnlocker client definition
+ * @returns {null}
+ */
+function generateWalletUnlockerClient ({ host, protoPath, tlsCertPath, logger }) {
+  const { lnrpc } = loadProto(protoPath)
+
+  if (!fs.existsSync(tlsCertPath)) {
+    throw new Error(`LND-ENGINE error - tls cert file not found at path: ${tlsCertPath}`)
+  }
+
+  const tls = fs.readFileSync(tlsCertPath)
+  const tlsCredentials = grpc.credentials.createSsl(tls)
+
+  try {
+    return new lnrpc.WalletUnlocker(host, tlsCredentials)
+  } catch (e) {
+    logger.error('Unable to create WalletUnlocker')
```

Comment:
what's the scenario in which we can't generate this client but we still want to be able to operate? I understand for the Lightning Client, but not the WalletUnlocker - shouldn't it always be available?

Reply:
They are interchangable. If the `Lightning` client is not avail, then WalletUnlocker is, but the RPC goes away once the wallet has been unlocked

Reply:
It seems like you should always be able to build these clients from the proto file even if the service is down, so this seems like a bad way to detect which service is being used.

Reply:
Sorry Trey, I was really wrong on this implementation. During development, there were a few issues I had that led me to this error, one of them being: missing deconstruction of `{ lnrpc }` which led to the handling of this error AND the previous implementation of `isUnlocked`. Please check out the latest commits for changes.